### PR TITLE
Add Activity Diagram

### PR DESCRIPTION
**Pull request by DonnyS57**
Added an Activity Diagram for DonnaFin

### CODE REVIEW

**Comment by sheehui** [comment]
lgtm donny, pls add this to the DG!

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -27,6 +27,13 @@ a Command Line Interface (CLI).
 
 ### 1.4 Typical User Workflow
 
+![Activity Diagram of DonnaFin](diagrams/ActivityDiagram.png)
+
+This Activity Diagram shows how one might navigate through the DonnaFin application in a typical usage.
```
Comment (sheehui) [inline docs/DeveloperGuide.md:32]: This Activity Diagram shows how one might typically navigate through the DonnaFin application.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -27,6 +27,13 @@ a Command Line Interface (CLI).
...
+
+
+Commands refer to the pre-defined functions that are used by the user. Invalid commands refer to commands that do not
+exist or commands that are used with the wrong format.
```
Comment (sheehui) [inline docs/DeveloperGuide.md:36]: Invalid commands refer to commands that are not available in the current window or commands that are used with the wrong format.

**Review by sheehui — COMMENTED** [review]
good work donny! nice diagram hehe. lmk what u think about the suggestions

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (bharathcs) [reply docs/DeveloperGuide.md:32]: I believe activity diagram should be lower case. Not a class name or a proper noun, soz

Comment (bharathcs) [reply docs/DeveloperGuide.md:32]: I believe activity diagram should be lower case. Not a class name or a proper noun, soz

Comment (bharathcs) [reply docs/DeveloperGuide.md:32]:
```suggestion
This activity diagram shows how one might typically navigate through the DonnaFin application.
```

**Review by bharathcs — DISMISSED** [review]
Diagram looks sick :)

**Review by bharathcs — APPROVED** [review]

**Pull request merged**